src/common/components/build-history/index.js

```diff
@@ -29,15 +30,19 @@ export const BuildHistory = (props) => {
 BuildHistory.propTypes = {
   repository: PropTypes.shape({
     owner: PropTypes.string,
-    name: PropTypes.string
+    name: PropTypes.string,
+    fullName: PropTypes.string
   }),
   success: PropTypes.bool,
   builds: React.PropTypes.arrayOf(React.PropTypes.object)
 };
 
-function mapStateToProps(state) {
-  const builds = state.snapBuilds.builds;
-  const success = state.snapBuilds.success;
+function mapStateToProps(state, ownProps) {
+  const { fullName } = ownProps.repository;
+
+  // get builds for given repo from the store or set default empty values
+  const repoBuilds = state.snapBuilds[fullName] || snapBuildsInitialStatus;
```

Comment:
Because we store builds per repository the 'initial state' per repo is undefined (until anything is stored in the store for given repo), so when reading data from the store, we need to provide some defaults.